client/extensions/woocommerce/app/products/product-form-details-card.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import FormLabel from 'components/forms/form-label';
+import FormTextArea from 'components/forms/form-textarea';
+import FormTextInput from 'components/forms/form-text-input';
+import CompactFormToggle from 'components/forms/form-toggle/compact';
+
+export default class ProductFormDetailsCard extends Component {
+
+	static propTypes = {
+		product: PropTypes.shape( {
+			id: PropTypes.isRequired,
+			type: PropTypes.string.isRequired,
+			name: PropTypes.string,
+		} ),
+		editProduct: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.setName = this.setName.bind( this );
+		this.setDescription = this.setDescription.bind( this );
+		this.toggleFeatured = this.toggleFeatured.bind( this );
+	}
+
+	setName( e ) {
```

Comment:
After looking at this, and our discussion, I think it's best to consider wrapping this up in a general-purpose Higher Order Component. However, I'd like more examples of code that will use it before we go there. So I added a TODO comment in here so we don't forget about it. We can revisit this after we've got some more edit stuff done to compare against.